java/pattern/constraint/TypeConstraint.java

```diff
@@ -565,8 +566,17 @@ public boolean isPlays() {
 
         @Override
         public String toString() {
-            return PLAYS.toString() + SPACE + roleType +
-                    (overriddenRoleType != null ? "" + SPACE + AS + SPACE + overriddenRoleType : "");
+            String returnString = PLAYS.toString() + SPACE + roleType;
+            if (overriddenRoleType != null) {
+                String overriddenRoleTypeString;
+                if (overriddenRoleType.label().isPresent()) {
+                    overriddenRoleTypeString = overriddenRoleType.label().get().label();
+                } else {
+                    overriddenRoleTypeString = overriddenRoleType.reference().syntax();
+                }
+                returnString += "" + SPACE + AS + SPACE + overriddenRoleTypeString;
+            }
+            return returnString;
```

Comment:
so if we don't have an established name for the string we build up and return, i think we can improve `returnString` to be something more meaningful, such as `syntax`